godot-core/src/meta/godot_convert/impls.rs

```diff
@@ -326,6 +326,61 @@ impl FromGodot for u64 {
     }
 }
 
+// ----------------------------------------------------------------------------------------------------------------------------------------------
+// Collections
+
+impl<T: ArrayElement> GodotConvert for Vec<T> {
+	type Via = Array<T>;
+}
+
+impl<T: ArrayElement> ToGodot for Vec<T> {
+	fn to_godot(&self) -> Self::Via {
+		Array::from(self.as_slice())
+	}
+}
+
+impl<T: ArrayElement> FromGodot for Vec<T> {
+	fn try_from_godot(via: Self::Via) -> Result<Self, ConvertError> {
+		Ok(via.iter_shared().collect())
+	}
+}
+
+impl<T: ArrayElement, const LEN: usize> GodotConvert for [T; LEN] {
+	type Via = Array<T>;
+}
+
+impl<T: ArrayElement, const LEN: usize> ToGodot for [T; LEN] {
+	fn to_godot(&self) -> Self::Via {
+		Array::from(self)
+	}
+}
+
+impl<T: ArrayElement, const LEN: usize> FromGodot for [T; LEN] {
+	fn try_from_godot(via: Self::Via) -> Result<Self, ConvertError> {
+		let vec = via.iter_shared().collect::<Vec<T>>();
+		
+		Self::try_from(vec)
+			.map_err(|vec| {
+				let message = format!(
+					"Array length mismatch, expected: `{}`, got: `{}`", LEN, vec.len()
+				);
+				
+				ConvertError::with_kind_value(ErrorKind::Custom(Some(message.into())), vec)
+			})
```

Comment:
This can, and ought, to be done without a collection.

```suggestion
	fn try_from_godot(via: Self::Via) -> Result<Self, ConvertError> {
        use std::mem::{ManuallyDrop, MaybeUninit};
        /// [`MaybeUninit::assume_init_array`] polyfill.
        ///
        /// # Safety
        ///
        /// It is up to the caller to guarantee that all elements of the array are
        /// in an initialized state.
        const unsafe fn aai<T, const N: usize>(array: [MaybeUninit<T>; N]) -> [T; N] {
            /// [`core::intrinsics::transmute_unchecked`] polyfill.
            /// # Safety
            ///
            /// Same as [`core::mem::transmute()`].
            const unsafe fn transmute_unchecked<T, U>(value: T) -> U {
                // Create union type that can store a `T` or a `U`.
                // We can then use this to convert between them.
                //
                // The repr(C) layout forces no offset between `t` and `u` as talked about here
                // https://rust-lang.github.io/unsafe-code-guidelines/layout/unions.html#c-compatible-layout-repr-c
                #[repr(C)]
                union Transmute<T, U> {
                    t: ManuallyDrop<T>,
                    u: ManuallyDrop<U>,
                }
                // Create the union in the `T` state.
                let value = Transmute {
                    t: ManuallyDrop::new(value),
                };
                // Read from the union in the `U` state.
                // SAFETY: This is safe because the caller has promised that `T` can be transmuted to `U`.
                // The following reference link talks about repr(C) unions being used this way.
                // https://doc.rust-lang.org/reference/items/unions.html#reading-and-writing-union-fields
                unsafe { ManuallyDrop::into_inner(value.u) }
            }
            
            // SAFETY: the caller must guarantee that `self` is initialized.
            unsafe { transmute_unchecked(array) }
        }

        // Make sure the length of the array matches what its copied from.
        if via.len() != LEN {
            let message = format!(
                "Array length mismatch, expected: `{}`, got: `{}`",
                LEN,
                via.len()
            );

            return Err(ConvertError::with_kind_value(
                ErrorKind::Custom(Some(message.into())),
                via,
            ));
        }

        // Create an uninit array for the output.
        let mut out = [const { MaybeUninit::uninit() }; LEN];
        // Copy each element from `via` to the returned value.
        for (item, copy) in via.iter_shared().by_ref().zip(&mut out) {
            copy.write(item);
        }
        // SAFETY: all items must have been written to, at this point.
        // if a panic has occurred, which I dont think it can, the items in `out` are leaked.
        Ok(unsafe { aai(out) })
```

Reply:
If you prefer not to write out all that unsafe you can use my `car` crate and just do [`car::try_from_array_option`](https://docs.rs/car/latest/car/macro.try_from_fn_option.html)`(|_| iter.next()).map_err(...)`

Reply:
@bend-n Thanks for this implementation! Since this is quite involved and may require its own review, I'd rather not push the responsibility on @Houtamelo for this change and instead suggest you later open a separate issue/PR where we can discuss this. 

If people agree that the extra unsafety is worth it, it probably makes sense to extract the polyfills to global functions, e.g. in `godot-ffi/src/toolbox.rs`. So they could be added independently of `GodotConvert` and also get their own unit tests.